prime-router/src/test/kotlin/SubmissionReceiverTests.kt

```diff
@@ -670,4 +707,80 @@ class SubmissionReceiverTests {
             queueMock.sendMessage(elrProcessQueueName, any())
         }
     }
+
+    @Test
+    fun `test ELR receiver validateAndMoveToProcessing, invalid message type hl7`() {
+        // setup
+        mockkObject(SubmissionReceiver.Companion)
+        val one = Schema(name = "one", topic = "test", elements = listOf(Element("a"), Element("b")))
+        val metadata = Metadata(schema = one)
+        val settings = FileSettings().loadOrganizations(oneOrganization)
+        val engine = makeEngine(metadata, settings)
+        val actionHistory = spyk(ActionHistory(TaskAction.receive))
+
+        val report = Report(
+            one,
+            mapOf<String, List<String>>(Pair("test", listOf("1,2"))),
+            source = ClientSource("ignore", "ignore"),
+            metadata = metadata
+        )
+
+        val receiver = spyk(
+            ELRReceiver(
+                engine,
+                actionHistory
+            )
+        )
+
+        val sender = CovidSender(
+            "Test Sender",
+            "test",
+            Sender.Format.HL7,
+            schemaName =
+            "one",
+            allowDuplicates = true
+        )
+        val actionLogs = ActionLogger()
+        val readResult = ReadResult(report, actionLogs)
+        val blobInfo = BlobAccess.BlobInfo(Report.Format.HL7, "test", ByteArray(0))
+        val routeResult = emptyList<ActionLog>()
+
+        every { engine.parseCovidReport(any(), any(), any()) } returns readResult
+        every { engine.recordReceivedReport(any(), any(), any(), any(), any()) } returns blobInfo
+        every { engine.routeReport(any(), any(), any(), any(), any()) } returns routeResult
+        every { engine.insertProcessTask(any(), any(), any(), any()) } returns Unit
+        every { queueMock.sendMessage(elrProcessQueueName, any()) } returns Unit
+
+        // act
+        var exceptionThrown = false
+        try {
+            receiver.validateAndMoveToProcessing(
+                sender,
+                hl7_record_bad_type,
+                emptyMap(),
+                Options.None,
+                emptyList(),
+                true,
+                true,
+                ByteArray(0),
+                "test.csv",
+                metadata = metadata
+            )
+        } catch (ex: IllegalArgumentException) {
+            exceptionThrown = true
+        }
+
+        // assert
+        assertTrue(exceptionThrown)
```

Comment:
Change this and remove the try above and use `assertThat {<your block of code>}.isFailure()`